File: apps/api/momento_api/models/album.py
from typing import Optional

from pydantic import BaseModel

from momento_api.models.media import MediaResponse


class AlbumResponse(BaseModel):
    id: int
    name: str
    description: Optional[str]
    cover_media_id: Optional[int]
    media_count: int
    created_at: str


class AlbumDetailResponse(BaseModel):
    id: int
    name: str
    description: Optional[str]
    cover_media_id: Optional[int]
    media: list[MediaResponse]
    created_at: str


class AlbumGetRequest(BaseModel):
    album_id: int


class AlbumCreateRequest(BaseModel):
    name: str
    description: Optional[str] = None


class AlbumUpdateRequest(BaseModel):
    album_id: int
    name: Optional[str] = None
    description: Optional[str] = None
    cover_media_id: Optional[int] = None


class AlbumDeleteRequest(BaseModel):
    album_id: int


class AlbumAddMediaRequest(BaseModel):
    album_id: int
    media_ids: list[int]


class AlbumRemoveMediaRequest(BaseModel):
    album_id: int
    media_ids: list[int]


class AlbumReorderRequest(BaseModel):
    album_id: int
    media_ids: list[int]


class AlbumListResponse(BaseModel):
    albums: list[AlbumResponse]

File: apps/api/src/models/album.rs
use crate::models::MediaResponse;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumResponse {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub cover_media_id: Option<i64>,
    pub media_count: i64,
    pub created_at: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumDetailResponse {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub cover_media_id: Option<i64>,
    pub media: Vec<MediaResponse>,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumGetRequest {
    pub album_id: i64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumCreateRequest {
    pub name: String,
    pub description: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumUpdateRequest {
    pub album_id: i64,
    pub name: Option<String>,
    pub description: Option<String>,
    pub cover_media_id: Option<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumDeleteRequest {
    pub album_id: i64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumAddMediaRequest {
    pub album_id: i64,
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumRemoveMediaRequest {
    pub album_id: i64,
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumReorderRequest {
    pub album_id: i64,
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AlbumListResponse {
    pub albums: Vec<AlbumResponse>,
}
